packages/equality/src/equality.ts

```diff
@@ -1,218 +1,11 @@
-const { toString, hasOwnProperty } = Object.prototype;
-const fnToStr = Function.prototype.toString;
-const previousComparisons = new Map<object, Set<object>>();
+import { DeepChecker } from "./checker";
 
 /**
  * Performs a deep equality check on two JavaScript values, tolerating cycles.
  */
 export function equal(a: any, b: any): boolean {
-  try {
-    return check(a, b);
-  } finally {
-    previousComparisons.clear();
-  }
+  return new DeepChecker().check(a, b);
```

Comment:
For better or worse, since equality checking now potentially calls into user-provided `deepEquals` methods, we cannot be sure the `check(a, b)` function will return before other code calls `equal(...)`, so the trick of using a single `previousComparisons` map and clearing it after each check no longer works, and we need to allocate a separate `DeepChecker` (with its own `comparisons` map) for each call to `equal(a, b)`.